services/datcore-adapter/src/simcore_service_datcore_adapter/modules/pennsieve.py

```diff
@@ -19,42 +19,75 @@
 logger = logging.getLogger(__name__)
 
 Total = int
-
-# NOTE: each pennsieve client seems to be about 8MB. so let's keep max 32
-@lru_cache(maxsize=32)
-def _create_pennsieve_client(api_key: str, api_secret: str) -> Pennsieve:
-    return Pennsieve(api_token=api_key, api_secret=api_secret)
-
-
-async def _get_pennsieve_client(api_key: str, api_secret: str) -> Pennsieve:
-    return await asyncio.get_event_loop().run_in_executor(
-        None,
-        lambda: _create_pennsieve_client(api_key=api_key, api_secret=api_secret),
-    )
-
-
-async def _get_authorization_headers(api_key: str, api_secret: str) -> dict[str, str]:
-    ps: Pennsieve = await _get_pennsieve_client(api_key=api_key, api_secret=api_secret)
-    bearer_code = ps._api.token  # pylint: disable=protected-access
-    return {"Authorization": f"Bearer {bearer_code}"}
+_GATHER_MAX_CONCURRENCY = 10
 
 
 def _compute_file_path(
     all_packages: dict[str, dict[str, Any]], pck: dict[str, Any]
 ) -> Path:
     file_path = Path(pck["content"]["name"])
     if "extension" in pck:
-        file_path = ".".join((f"{file_path}", pck["extension"]))
+        file_path = Path(".".join((f"{file_path}", pck["extension"])))
     if parent_id := pck["content"].get("parentId"):
         file_path = (
             _compute_file_path(all_packages, all_packages[parent_id]) / file_path
         )
     return Path(file_path)
 
 
+class PennsieveAuthorizationHeaders(TypedDict):
+    Authorization: str
+
+
+_TTL_CACHE_AUTHORIZATION_HEADERS_SECONDS: Final[
+    int
+] = 3530  # NOTE: observed while developing this code, pennsieve authorizes 3600 seconds, so we cache a bit less
```

Comment:
Question: what happens if this expires?

Reply:
good catch. I think I need to rework that thing

Reply:
done